### ISSUE TITLE
`no-named-as-default` should not report on different types

### ISSUE THREAD

**Issue by JounQin** [un-ts/eslint-plugin-import-x#312]
```ts
import Immutable from 'seamless-immutable'; // `Immutable` is a `constructor`
```

```ts
import { Immutable } from 'seamless-immutable'; // `Immutable` is a `type`
```

Workaround:

```ts
import SeamlessImmutable from 'seamless-immutable';
```

But this is clearly not preferred.

**Comment by silverwind** [comment]
This might be a non-issue when using https://www.typescriptlang.org/tsconfig/#verbatimModuleSyntax in Typescript, which would flag the missing `import type` statement for the second example. I think it's good practice to force `import type` syntax for type imports.

**Comment by JounQin** [comment]
>  I think it's good practice to force `import type` syntax for type imports.

It's indeed a goog practice, but not everyone is using it, and older TypeScript versions doesn't support `verbatimModuleSyntax` neither.
